### ISSUE TITLE
Considerations and assumptions

### ISSUE THREAD

**Issue by chrdebru** [kg-construct/rml-io#55]
- [ ] When CSVW header is false, I assume that columns are numbered from 1 to n (as CSVW assumes column numbers start from 1). Is this OK?
- [ ] We have `rml:null` and `csvw:null`. Which takes precedence?

**Comment by DylanVanAssche** [comment]
> When CSVW header is false, I assume that columns are numbered from 1 to n (as CSVW assumes column numbers start from 1). Is this OK?

CSVW supports that but RML does not at this point, a header is always currently required.
This is kinda of a corner-case we have to define I suppose?

> Allow RDF terms for rml:null for resources based on RDF and SPARQL?

For SPARQL there is things like `EXISTS` and such right?
`rml:null` is actually only for access descriptions that don't define what NULL values are.

**Comment by bjdmeest** [comment]
rml:null _does_ define what NULL values are, no? that's my preference and how I interpret the spec: https://kg-construct.github.io/rml-io/spec/docs/#null-values. So then I would agree with the suggestion to extend to RDF terms

**Comment by DylanVanAssche** [comment]
Yes, you can use it for any source to indicate what NULL values are, extending to RDF etc. was already assumed, but not said explicitely?

**Comment by DylanVanAssche** [comment]
@chrdebru 

> When CSVW header is false, I assume that columns are numbered from 1 to n (as CSVW assumes column numbers start from 1). Is this OK?

How do you think we can define this better? We always need it for making references.

>  Allow RDF terms for rml:null for resources based on RDF and SPARQL?

This cannot be solved currently as RDF as Source is being discussed and probably for WG. Will split off the issue as we have 2 things now in 1 issue.
